pkg/performanceprofile/controller/performanceprofile/hypershift/hypershift.go

```diff
@@ -129,13 +129,17 @@ func EncodeManifest(obj runtime.Object, scheme *runtime.Scheme) ([]byte, error)
 	return buff.Bytes(), err
 }
 
-func DecodeManifest(b []byte, scheme *runtime.Scheme, obj runtime.Object) error {
+// DecodeManifest tries to decode b into obj.
+// if fails to decode b, it returns an error
+// if decoded data returned from b, has different GVK than what stored in obj
+// it returns false
```

Comment:
is this a legit use case of the code or are we abusing some behavior?

Reply:
This abstraction based on the documentation of the underlying `decode` function:
https://github.com/openshift/cluster-node-tuning-operator/blob/32e3df6b94d52dabbd44e82d86bc3e2798b2c616/vendor/k8s.io/apimachinery/pkg/runtime/serializer/json/json.go#L127-#L134

Reply:
nice, TIL